### PR TITLE
relax strict test for bytes type

### PR DESCRIPTION
**Pull request by Midnighter**
This is a possible solution to #683. I'm not sure if you prefer to keep an `assert` here or if it conforms with your general coding style.

### CODE REVIEW

**Comment by RonnyPfannschmidt** [comment]
at first glance the errors on Travis are unrelated

**Comment by untitaker** [comment]
I also learn towards keeping it an assertion.

**Comment by Midnighter** [comment]
OK, changed and rebased.

File: werkzeug/serving.py [hunk 1]
```diff
@@ -148,7 +148,11 @@ def write(data):
                     self.send_header('Date', self.date_time_string())
                 self.end_headers()
 
-            assert type(data) is bytes, 'applications must write bytes'
+            assert isinstance(data, bytes),\
+                    'applications must write bytes - found \'%s\'' % type(data)
+# relax the strict `type(data) is bytes` assertion in order to allow subclasses
```
Comment (untitaker) [inline werkzeug/serving.py]: This comment is badly indented... I think we should omit it.

Comment (RonnyPfannschmidt) [inline werkzeug/serving.py]: the idea of that assertion is that it only fires when you do something thats broken,
as far as i'm concerned extra type information is not necessary at that place

**Comment by Midnighter** [comment]
Alright, this is now the minimal change. I've removed the comment and rather extended the commit message.

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**